test/Chainweb/Test/Pact/SPV.hs

```diff
@@ -1,183 +1,187 @@
+
+            -- 'S.effects' forces the stream here. It -must- occur so that we evaluate the stream
+            --
+            void $! S.effects $ extendTestCutDb cutDb pact1 60
+            syncPact cutDb pact1
+
+            -- execute '(coin.create-coin ...)' using the  correct chain id and block height
+            pact2 <- testWebPactExecutionService v (Just cdb) $ txGenerator2 cdb cid1 bh1
+            syncPact cutDb pact2
+
+            -- if we get this far, we have succeeded
+            void $! S.head_ $ extendTestCutDb cutDb pact2 1
 
--- Find a reachable target chain
---
-targetChain :: Cut -> BlockHeader -> IO ChainId
-targetChain c srcBlock = do
-    cids <- generate (shuffle $ toList $ chainIds c)
-    go cids
   where
-    graph = _chainGraph c
-
-    go [] = error
-        $ "SPV proof test failed to find a reachable target chain. This is a bug in the test code"
-        <> ". source block: " <> sshow srcBlock
-        <> ". current cut: " <> sshow c
-    go (h:t) = if isReachable h then return h else go t
-
-    chainHeight trgChain = _blockHeight (c ^?! ixg trgChain)
-
-    isReachable trgChain
-        = _blockHeight srcBlock <= chainHeight trgChain - distance trgChain
-
-    distance x = len $ shortestPath (_chainId srcBlock) x graph
+    v = TimedCPM petersonChainGraph
+    logg l
+        | l <= Warn = T.putStrLn . logText
+        | otherwise = const $ return ()
 
-runRST :: Monad m => ReaderT r (StateT s m) a -> r -> s -> m a
-runRST action pse st = fmap fst $
-  runStateT (runReaderT action pse) st
 
--- -------------------------------------------------------------------------- --
--- Pact Service setup
+type TransactionGenerator
+    = ChainId -> BlockHeight -> BlockHash -> IO (Vector ChainwebTransaction)
 
+-- | Generate burn/create Pact Service commands
+--
+txGenerator1 :: TransactionGenerator
+txGenerator1 _cid _bhe _bha = do
+    mkPactTestTransactions' txs
+  where
+    txs =
+      let c =
+            [text|
+              (coin.delete-coin 'sender00 1 'sender01 (read-keyset 'sender01-keys) 1.0)
+              |]
 
-withPactSetup :: CutDb cas -> (PactServiceEnv -> PactDbState -> IO a) -> IO a
-withPactSetup cdb f = do
-    let l = newLogger alwaysLog (LogName "pact-spv")
-        genv = GasEnv 0 0.0 (constGasModel 0)
-
-    mv <- newMVar cdb
-    (cpe, st) <- initConf l genv
-    void $ saveInitial (cpe ^. cpeCheckpointer) st
-
-    let spv = pactSpvSupport mv
+      in fromList [ PactTransaction c keys ]
 
-    let pse = PactServiceEnv Nothing cpe spv def
-        pss = PactServiceState st Nothing
 
-    initCC pse pss >> f pse st
+-- | Generate the 'create-coin' command in response
+-- to the previous 'delete-coin' call
+--
+txGenerator2
+    :: PayloadCas cas
+    => MVar (CutDb cas)
+    -> ChainId
+    -> BlockHeight
+    -> TransactionGenerator
+txGenerator2 cdbv cid bhe tid _bhe _bha =
+    if tid == unsafeChainId 1
```

Comment:
as a naive first test, the strategy is detailed in the documents. Additionally, we ensure that the `create-coin` command is only ever issued on the correct chain by doing this branch. Ideally, in the future, we'd have a smarter tx generator which would generate these tests and monitor meta-information like target/source chainid pairs